### PR TITLE
refactor K8s docs for v21.2

### LINKED ISSUE #11109: Several "Orchestrate Kubernetes" docs in v21.2 refer to much older versions (e.g. 20.2) in upgrade steps
Specifically, the [upgrade the cluster](https://www.cockroachlabs.com/docs/dev/orchestrate-cockroachdb-with-kubernetes-multi-cluster.html#upgrade-the-cluster) section refers to some very old versions and probably needs editing.   Also the one [on this page](https://www.cockroachlabs.com/docs/dev/operate-cockroachdb-kubernetes.html#upgrade-the-cluster) - maybe an include?

Editing the steps would require changing enough information that I wasn't comfortable doing it in a mechanical way as part of the v21.2 docs version bump in #11035, since it looks complicated enough to probably need some re-testing (or at least, it looks that way to someone who doesn't know this area so I didn't want to just do it)

### LINKED ISSUE #11662: EKS-Anywhere new callout - Deploy CockroachDB in a Single Kubernetes Cluster Doc Update
Re: [Deploy CockroachDB in a Single Kubernetes Cluster](https://cockroachlabs.com/docs/v21.1/deploy-cockroachdb-with-kubernetes.html)

## Issue Description

Need to add a callout to the EKS setup section that if running EKS-Anywhere, CockroachDB requires that a default storage class that can auto-provision PVs is configured. Alternatively, customers can define a custom storage config as required by their install pattern. Otherwise, EKS-A users can simply follow the default EKS-Anywhere setup instructions for building their cluster and then skip to step 2.

## Suggested Resolution 

A quick "Note" box in the start kubernetes section should be sufficient.

### PR DESCRIPTION
**Pull request by taroface**
- Refactored the K8s documentation for v21.2 to cover db operator tasks in separate articles. This should be easier to update and maintain.
- Included changes from #11206.
- Also fixes #11109 for all three deployment methods, including some outdated parts in v21.1.
- Fixes #11662.

### CODE REVIEW

**Comment by taroface** [comment]
@udnay This is largely a restructure of the docs to make them more usable/navigable. There aren't really any content changes here, but let me know if you have thoughts on the organization.

File: _includes/sidebar-data-v21.2.json [hunk 1]
```diff
@@ -803,50 +803,78 @@
...
+                    "title": "Multi-Cluster Deployment",
+                    "urls": [
+                      "/${VERSION}/orchestrate-cockroachdb-with-kubernetes-multi-cluster.html"
+                    ]
+                  }
                 ]
               },
               {
-                "title": "Docker Swarm Deployment",
+                "title": "Docker Swarm",
```
Comment (udnay) [inline _includes/sidebar-data-v21.2.json:877]: Are we planning on deprecating this soon? I'm not sure who supports this. /cc @keith-mcclellan

File: _includes/v21.1/orchestration/start-cockroachdb-helm-secure.md [hunk 2]
```diff
@@ -34,32 +34,32 @@ Secure CockroachDB deployments on Amazon EKS via Helm are [not yet supported](ht
...
-            ~~~
+        {% include_cached copy-clipboard.html %}
+        ~~~
+        conf:
+          cache: "2Gi"
+          max-sql-memory: "2Gi"
+        ~~~
 
-        1. For a secure deployment, set `tls.enabled` to `true`.
+    1. For a secure deployment, set `tls.enabled` to `true`.
```
Comment (udnay) [inline _includes/v21.1/orchestration/start-cockroachdb-helm-secure.md:52]: Do we have docs for the self signer in here as well?

File: _includes/v21.2/orchestration/start-cockroachdb-operator-secure.md [hunk 3]
```diff
@@ -64,7 +60,11 @@ By default, the Operator will generate and sign 1 client and 1 node certificate
     ~~~
 
     {{site.data.alerts.callout_info}}
-    By default, this manifest specifies CPU and memory resources that are appropriate for the virtual machines used in this deployment example. On a production cluster, you should [substitute values](operate-cockroachdb-kubernetes.html#allocate-resources) that are appropriate for your machines and workload.
+    By default, this custom resource specifies CPU and memory resources that are appropriate for the virtual machines used in this deployment example. On a production cluster, you should substitute values that are appropriate for your machines and workload. For details on configuring your deployment, see [Configure the Cluster](configure-cockroachdb-kubernetes.html).
+    {{site.data.alerts.end}}
+
+    {{site.data.alerts.callout_info}}
+    By default, the Operator will generate and sign 1 client and 1 node certificate to secure the cluster. To authenticate using your own CA, see [Secure the Cluster](secure-cockroachdb-kubernetes.html).
```
Comment (udnay) [inline _includes/v21.2/orchestration/start-cockroachdb-operator-secure.md:67]: Maybe add a note that we use a `cockroach` generated CA if none is provided.

**Review by udnay — APPROVED** [review]
Couple of nits but LGTM

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11623)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @keith-mcclellan and @udnay)

___
*[_includes/sidebar-data-v21.2.json, line 877 at r3](https://reviewable.io/reviews/cockroachdb/docs/11623#-MjBcE2qqWrauaVjK6dH-r3-877:-MkIqhGe5Q-qj5BGR5Xy:bymurgj) ([raw file](https://github.com/cockroachdb/docs/blob/81b71c7350fef1515e19e1b87fda69c93ced074f/_includes/sidebar-data-v21.2.json#L877)):*
<details><summary><i>Previously, udnay (Yandu Oppacher) wrote…</i></summary><blockquote>

Are we planning on deprecating this soon? I'm not sure who supports this. /cc @keith-mcclellan 
</blockquote></details>

Not sure, but we can handle this in a separate ticket if necessary.
___
*[_includes/v21.1/orchestration/start-cockroachdb-helm-secure.md, line 52 at r3](https://reviewable.io/reviews/cockroachdb/docs/11623#-MjBcE2qqWrauaVjK6dK-r3-52:-MkIqkUM0Z3ztbDKnlui:b-7xmeq5) ([raw file](https://github.com/cockroachdb/docs/blob/81b71c7350fef1515e19e1b87fda69c93ced074f/_includes/v21.1/orchestration/start-cockroachdb-helm-secure.md#L52)):*
<details><summary><i>Previously, udnay (Yandu Oppacher) wrote…</i></summary><blockquote>

Do we have docs for the self signer in here as well?
</blockquote></details>

This should include everything that was in the PR for the self-signer. My understanding is that `tls.enabled=true` enables the self-signer; let me know if this isn't correct.
___
*[_includes/v21.2/orchestration/start-cockroachdb-operator-secure.md, line 67 at r3](https://reviewable.io/reviews/cockroachdb/docs/11623#-MjBcE2qqWrauaVjK6dR-r3-67:-MkIvQ7k7n_03oSFDrRF:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/81b71c7350fef1515e19e1b87fda69c93ced074f/_includes/v21.2/orchestration/start-cockroachdb-operator-secure.md#L67)):*
<details><summary><i>Previously, udnay (Yandu Oppacher) wrote…</i></summary><blockquote>

Maybe add a note that we use a `cockroach` generated CA if none is provided.
</blockquote></details>

Done.

**Comment by taroface** [comment]
@kathancox This is a large update that mostly restructures existing content. See my comment here: https://github.com/cockroachdb/docs/pull/11623#issuecomment-916990752

**Review by udnay — APPROVED** [review]
> Reviewed 21 of 24 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11623)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox, @keith-mcclellan, and @udnay)

File: v21.2/orchestrate-cockroachdb-with-kubernetes-multi-cluster.md [hunk 4]
```diff
@@ -1040,21 +1040,21 @@ Each of your Kubernetes clusters contains 3 instances that can run CockroachDB p
...
+The upgrade process on Kubernetes is a [staged update](https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#staging-an-update) in which the Docker image is applied to the pods one at a time, with each pod being stopped and restarted in turn. This is to ensure that the cluster remains available during the upgrade.
 
 1. Verify that you can upgrade.
 
-    To upgrade to a new version, you must first be on a production release of the previous version. The release does not need to be the latest production release of the previous version, but it must be a production release rather than a testing release (alpha/beta).
+    To upgrade to a new major version, you must first be on a production release of the previous version. The release does not need to be the latest production release of the previous version, but it must be a production release and not a testing release (alpha/beta).
```
Comment (kathancox) [inline v21.2/orchestrate-cockroachdb-with-kubernetes-multi-cluster.md:1049]: Same here, if you agree!

**Review by kathancox — APPROVED** [review]
LGTM! Just one small non-blocking comment :-)

**Comment by taroface** [comment]
@keith-mcclellan Can you let me know if this commit https://github.com/cockroachdb/docs/pull/11623/commits/692e85ef35ae1217e75fdbbac62976374bccb65e does it for the EKS-A callout?

**Pull request merged**